pallets/dispatcher/src/lib.rs

```diff
@@ -169,6 +178,62 @@ pub mod pallet {
 			AaveManagerAccount::<T>::put(account);
 			Ok(())
 		}
+
+		/// Dispatch a call with extra gas.
+		///
+		/// This allows executing calls with additional weight (gas) limit.
+		/// The call will fail if it would consume more weight than the specified limit.
+		#[pallet::call_index(3)]
+		#[pallet::weight({
+			let call_weight = call.get_dispatch_info().weight;
+			let call_len = call.encoded_size() as u32;
+			let gas_weight = T::GasWeightMapping::gas_to_weight(*extra_gas, true);
+			T::WeightInfo::dispatch_with_extra_gas(call_len)
+				.saturating_add(call_weight)
+				.saturating_add(gas_weight)
+		})]
+		pub fn dispatch_with_extra_gas(
+			origin: OriginFor<T>,
+			call: Box<<T as Config>::RuntimeCall>,
+			extra_gas: u64,
+		) -> DispatchResultWithPostInfo {
+			let who = ensure_signed(origin)?;
+
+			// Store the gas limit for this account
+			ExtraGas::<T>::set(extra_gas);
+
+			// Dispatch the call
+			let (result, actual_weight) = Self::do_dispatch(who.clone(), *call);
+
+			let remaining_extra_gas = ExtraGas::<T>::take();
+			let gas_used = extra_gas.saturating_sub(remaining_extra_gas);
+
+			// if gas used is zero, we can return the result directly
+			if gas_used == 0 {
+				return result;
+			}
+			// we can return result directly also when the actual weight is None ( for some reason)
+			let Some(actual_weight) = actual_weight else {
+				return result;
+			};
+			//Otherwise, we need to account for the extra gas used.
+			let extra_weight = T::GasWeightMapping::gas_to_weight(gas_used, true);
+			let actual_weight = Some(actual_weight.saturating_add(extra_weight));
+
+			match result {
+				Ok(call_post_info) => Ok(PostDispatchInfo {
+					actual_weight,
+					pays_fee: call_post_info.pays_fee,
```

Comment:
i think you should always pay extra gas, inner tx can waive the fee, but it shouldnt be able to do it for wrapper as well

Reply:
changed to Pays:Yes